Automattic/wpcom-proxy-request
# Detect metaAPI responses and simplify ready event

Requires patch D6466.

This PR fixes `wpcom.request( { metaAPI: ... } )` calls returning a rejected promise since #20. This is because this "special" request doesn't actually trigger an XHR, it simply updates a variable in the iframe context. Thus when returning it doesn't have a `statusCode`.
We're explicitly handling this case here by setting the `statusCode` to `200` when the correct body message (`metaAPIupdated`) is returned.

It also updates the `ready` event and expects a plain string in all cases.

### Testing Instructions
- Apply the server patch
- modify wp-calypso `package.json`:
```
-    "wpcom-proxy-request": "3.0.0",
+    "wpcom-proxy-request": "Automattic/wpcom-proxy-request#update\/metaapi",
```
- Start calypso with `make run` (`wpcom-proxy-request` will update)
- Check that there is no error in the console when you visit http://calypso.localhost:3000/

### Reviews
